products-service/handler-unique.js

```diff
@@ -1,30 +1,45 @@
 'use strict';
-import productList from './statics/productList.json';
+import { client } from './db-connection';
 
 const headers = {
   'Content-Type': 'application/json',
   'Access-Control-Allow-Origin': '*',
 };
 
 const getProductById = async (event) => {
-  const id = event.queryStringParameters.id;
-  const selectedProduct = productList.find(function (product) {
-    return id === product.id;
-  });
-  const [statusCode, product] =
-    typeof selectedProduct === 'undefined'
-      ? [404, 'Product not Found']
-      : [200, selectedProduct];
+  try {
+    console.log(
+      'Getting a unique product by id - on  GET /products/{id} endpoint'
+    );
+    const id = event.queryStringParameters.id;
+    const result = await client.query(getQueryString(id));
 
-  const body = JSON.stringify({
-    product: product,
-  });
+    const [statusCode, product] =
+      result.rows.length === 0
+        ? [404, 'Product not Found']
+        : [200, result.rows[0]];
 
-  return {
-    statusCode,
-    headers,
-    body,
-  };
+    const body = JSON.stringify({
+      product: product,
+    });
+
+    return {
+      statusCode,
+      headers,
+      body,
+    };
+  } catch (error) {
+    return {
+      statusCode: 500,
+      headers,
+      body: 'Server Error',
+    };
+  }
+};
+
+const getQueryString = (id) => {
+  return `SELECT products.*, stocks.product_count as count FROM products JOIN stocks 
+ON products.id = stocks.product_id AND products.id = '${id}';`;
```

Comment:
this approach is a bad practice. We should always remember about possible sql injections